README.md

```diff
@@ -32,31 +32,418 @@ There's a very unstable version of the plugin available to try out. It's held to
 
 `yarn rw dev`
 
-## API Reference
+> **Note**
+>
+> The following steps won't be needed for much longer
 
-After setting up your app via the CLI, you'll find a demo page has been added to your app. This is intended as an in-app API reference, though in a proper store it might a little different. 
+5. Checkout a subscription item using "user@test.com" as the email.
 
-### `<StripeProvider/>`
+6. Configure Customer Portal in the [Stripe Dashboard](https://dashboard.stripe.com/subscriptions) 
 
-### `useStripeCheckout`
+7. Play around!
+
+## Web-side API Reference
+
+After setting up your app via the setup command, you'll see that a demo page has been added to your app. This is intended as an in-app API reference—I'm assuming you would want your commerce project to look a different.
+
+### `<StripeProvider customer/>`
+
+Uses React context to manage the StripeCart, store customer data and contain the plugin logic on the web side.
+
+**Example 1**
+
+```jsx
+import { StripeProvider } from 'redwoodjs-stripe/web'
+
+// ...
+
+ <StripeProvider
+        customer={{
+          search: isLoggedIn ? `email: "${emailFromAuth}"` : '',
+        }}
+      >
+        <Storefront/>
+  </StripeProvider>
+```
+
+**Example 2**
+
+Routes.js
+```jsx
+import { StripeProvider } from 'redwoodjs-stripe/web'
+
+// ...
+
+<Set wrap={StripeProvider} customer={{search: isLoggedIn ? `email: "${emailFromAuth}"` : ''}}>
+  <Route path="/category" page={CategoryPage} name="category" />
+  <Route path="/product" page={ProductPage} name="product" />
+  // ...
+</Set>
+```
+
+`customer` has the following shape:
+```ts
+type Customer = {
+  search: string | ''
+}
+```
+`search` accepts a [querystring](https://stripe.com/docs/search#search-query-language) that is used to search for a Stripe Customer. The idea is to use data of your authenticated user to build the querystring. In most cases your authentication provider will provide you with an email address that you will be able to use to map to a Stripe Customer.
+
+### `useStripeCheckout()`
```

Comment:
Let's not use the function name itself as a section title as it's hard to grok. Try something like this:

> ### Stripe Checkout hook
> Redirect to Stripe's hosted Checkout by using the `useStripeCheckout` hook.